src/Component/Customer/CustomerSelector.php

```diff
@@ -29,26 +31,48 @@ class CustomerSelector implements CustomerSelectorInterface
     protected $session;
 
     /**
-     * @var \Symfony\Component\Security\Core\SecurityContextInterface
+     * @var TokenStorageInterface|SecurityContextInterface
      */
-    protected $securityContext;
+    protected $tokenStorage;
+    
+    /**
+     * @var AuthorizationCheckerInterface|SecurityContextInterface
+     */
+    protected $authorizationChecker;
 
     /**
      * @var string
      */
     protected $locale;
 
     /**
-     * @param CustomerManagerInterface $customerManager
-     * @param SessionInterface         $session
-     * @param SecurityContextInterface $securityContext
-     * @param LocaleDetectorInterface  $localeDetector
+     * NEXT_MAJOR: Go back to type hinting check when bumping requirements to SF 2.6+.
+     *
+     * @param CustomerManagerInterface                          $customerManager
+     * @param SessionInterface                                  $session
+     * @param TokenStorageInterface|SecurityContextInterface    $tokenStorage
+     * @param LocaleDetectorInterface                           $localeDetector
      */
-    public function __construct(CustomerManagerInterface $customerManager, SessionInterface $session, SecurityContextInterface $securityContext, LocaleDetectorInterface $localeDetector)
+    public function __construct(CustomerManagerInterface $customerManager, SessionInterface $session, $tokenStorage, $authorizationChecker, LocaleDetectorInterface $localeDetector)
```

Comment:
I added a BC break here the same as was done in https://github.com/sonata-project/SonataUserBundle/blob/6f37cdf9dc57c0c71ba3b01a1856dfd6f9c686fc/Security/EditableRolesBuilder.php
If you have another solution to not add a BC break I will fix